core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -860,17 +873,16 @@ object ContractInfoV0TLV extends TLVFactory[ContractInfoV0TLV] {
   }
 }
 
-case class TLVPoint(
-    outcome: Long,
-    value: Satoshis,
-    extraPrecision: Int,
-    isEndpoint: Boolean)
-    extends NetworkElement {
+sealed trait TLVPoint extends NetworkElement {
```

Comment:
I'm relatively indifferent here, but I think this is maybe the wrong approach for a type that is explicitly meant to represent a TLV, we have our internal point type separate from this one for the purpose of allowing internal representation but this external representation type should probably more directly reflect what we see over the wire (which is a bool flag). No operations are done without converting these to internal types, the only logic applied to these is serialization and de-serialization which is made marginally more convoluted by this